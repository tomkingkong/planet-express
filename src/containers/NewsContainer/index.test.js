import React from 'react';
import { shallow } from 'enzyme';

import { 
  NewsContainer, 
  mapStateToProps } from '.';

describe('News Container', () => {
  it('should match snapshot', () => {
    const mockProps = {
      news: [{}]
    };
    const wrapper = shallow(<NewsContainer {...mockProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  describe('mapStateToProps', () => {
    it('should be provided with a news array from props', () => {
      const mockState = {
        exoplanets: [],
        spaceEvents: [{}, {}],
        news: [],
        journey: {},
        technologies: []
      };
      const expected = { news: [] };
      const results = mapStateToProps(mockState);
      expect(results).toEqual(expected);
    });
  });
});

