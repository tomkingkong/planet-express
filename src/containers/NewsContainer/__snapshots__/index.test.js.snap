// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`News Container should match snapshot 1`] = `
<section
  className="news_container"
>
  <h1
    className="news__section__title"
  >
    Out of this world News!
  </h1>
  <Component
    key="0"
  />
</section>
`;
